src/message.rs

```diff
@@ -62,3 +64,57 @@ impl fmt::Display for Message {
         )
     }
 }
+
+impl Message {
+    pub fn to_annotated_source(&self) -> String {
+        let source = fs::read_to_string(&self.filename).unwrap();
+        let error_lines = source
+            .lines()
+            .skip(self.location.row() - 1)
+            .take(self.end_location.row() - self.location.row() + 1)
+            .collect::<Vec<_>>();
+        let body = self.kind.body();
+        let code = self.kind.code().as_ref();
+        let rel_path = relativize_path(Path::new(&self.filename));
+        let source = error_lines.join("\n");
+        let range = (
+            self.location.column() - 1,
+            error_lines
+                .iter()
+                .enumerate()
+                .map(|(i, line)| {
+                    if i == error_lines.len() - 1 {
+                        self.end_location.column()
+                    } else {
+                        line.len() + 1
+                    }
+                })
+                .sum::<usize>()
+                - 1,
+        );
+        let snippet = Snippet {
+            title: Some(Annotation {
+                label: Some(&body),
+                id: Some(code),
+                annotation_type: AnnotationType::Error,
+            }),
+            footer: vec![],
+            slices: vec![Slice {
+                source: &source,
+                line_start: self.location.row(),
+                origin: Some(&rel_path),
+                fold: false,
+                annotations: vec![SourceAnnotation {
+                    label: "",
+                    annotation_type: AnnotationType::Error,
+                    range,
+                }],
+            }],
+            opt: FormatOptions {
+                color: true,
+                ..Default::default()
+            },
+        };
+        DisplayList::from(snippet).to_string()
```

Comment:
I think that if we're going to use this crate for `--show-source`, we should probably use the same crate and style even when `--show-source` is false (and just omit the source code). What do you think?

Reply:
Sounds good. I think we can set `slices` to an empty vector when `--show-source` is false.